src/systems/filecoin_files/data_transfer/data_transfer_subsystem.id

```diff
@@ -1,66 +1,127 @@
-type Address struct {}
-type Storage struct {}
-type StorageDealID struct {}
-type Function struct {}
-type Piece struct {}
+import ipld "github.com/filecoin-project/specs/libraries/ipld"
+import libp2p "github.com/filecoin-project/specs/libraries/libp2p"
 
-type DataTransferSubsystem struct @(mutable) {
-    dataTransfers  {UInt: DataTransferChannel}
-    volumes        [Storage]
+import piece "github.com/filecoin-project/specs/systems/filecoin_files/piece"
+
+type StorageDeal struct {}
+type RetrievalDeal struct {}
+
+// A DataTransferVoucher is used to validate
+// a data transfer request against the underlying storage or retrieval deal
+// that precipitated it
+type DataTransferVoucher union {
+	StorageDealVoucher 
+	RetrievalDealVoucher
+}
+
+type StorageDealVoucher struct {
+	deal StorageDeal
+}
+
+type RetrievalDealVoucher struct {
+	deal RetrievalDeal 
+}
```

Comment:
- i think these vouchers should probably be defined in the Storage Market area. (there may be an annoying import problem to make the `DataTransferVoucher` union --  not sure if `id` takes `interface{}` yet. 
- Could be defined in `data_transfer_subsystem.go` as an interface